compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3709,13 +3709,22 @@ boolean validNumericTypeExists(BType type) {
                             if (!checkValueSpaceHasSameType((BFiniteType) memType, baseExprType)) {
                                 return false;
                             }
+                            continue;
                         }
-                        if (!checkValidNumericTypesInUnion(memType, firstTypeInUnion.tag)) {
+                        if (!checkValidNumericTypesInUnion(memType, baseExprType.tag)) {
```

Comment:
The previous check is trying to if all the members of the union have the same type. Now for finite types, we don't do this. Check below code if even with `D` whether we should allow the operator
```Ballerina
const decimal A = 1.25;
type C 3.45d|1.25d|45.6d;
type D "Me";
type des C|decimal|2.0d|2.0d;

type DecimalType1 A|des|C|D;


public function main() {
    A a = 1.25;
    decimal|DecimalType1 b = 25346.45d;

    decimal|string x = a + b;
}
```

Reply:
With `D`, operator is not allowed

Reply:
This is fixed now